File: codegen/results/v001/gemma4_26b_think/20260407_160800/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"os"
	"strconv"
)

// Person represents the structure of each row in the CSV
type Person struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	// Open the target CSV file
	file, err := os.Open("input/data.csv")
	if err != nil {
		// If the file cannot be opened, we exit without outputting to stdout
		// to avoid violating the "Output must be valid JSON" constraint.
		return
	}
	defer file.Close()

	reader := csv.NewReader(file)

	// Read all records from the CSV
	records, err := reader.ReadAll()
	if err != nil {
		return
	}

	// If there are no records or only a header, output an empty array
	if len(records) <= 1 {
		json.NewEncoder(os.Stdout).Encode([]Person{})
		return
	}

	// Skip the header (index 0) and process data rows
	dataRows := records[1:]
	var people []Person

	for _, row := range dataRows {
		// Ensure the row has enough columns to prevent index out of range
		if len(row) < 4 {
			continue
		}

		// Parse Age (string to int)
		age, err := strconv.Atoi(row[1])
		if err != nil {
			age = 0
		}

		// Parse Score (string to float64)
		score, err := strconv.ParseFloat(row[3], 64)
		if err != nil {
			score = 0.0
		}

		// Construct the Person object
		person := Person{
			Name:  row[0],
			Age:   age,
			Email: row[2],
			Score: score,
		}

		people = append(people, person)
	}

	// Encode the slice of Person objects directly to stdout as a JSON array
	encoder := json.NewEncoder(os.Stdout)
	err = encoder.Encode(people)
	if err != nil {
		return
	}
}

File: codegen/results/v001/gemma4_26b_think/20260407_160800/typescript/001_csv_to_json/solution.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * This script reads a CSV file located at 'input/data.csv'
 * and converts its rows into a JSON array of objects.
 * 
 * Expected CSV Format:
 * Name,Age,Email,Score
 * John Doe,30,john@example.com,85.5
 * 
 * Output:
 * [
 *   {
 *     "Name": "John Doe",
 *     "Age": 30,
 *     "Email": "john@example.com",
 *     "Score": 85.5
 *   }
 * ]
 */

function main() {
  const filePath = path.join(process.cwd(), 'input', 'data.csv');

  try {
    // Check if file exists to prevent throwing an unhandled error to stderr
    // which might violate the "Do not output anything else" constraint.
    if (!fs.existsSync(filePath)) {
      process.stdout.write('[]');
      return;
    }

    const fileContent = fs.readFileSync(filePath, 'utf-8');
    
    // Split into lines and remove empty lines or whitespace-only lines
    const lines = fileContent
      .split(/\r?\n/)
      .map(line => line.trim())
      .filter(line => line.length > 0);

    // If there is no data or only a header, return empty array
    if (lines.length <= 1) {
      process.stdout.write('[]');
      return;
    }

    // Skip the header row (index 0) and map the rest
    const jsonData = lines.slice(1).map(line => {
      const columns = line.split(',');

      // Mapping based on the requirements:
      // Name: string, Age: integer, Email: string, Score: float
      return {
        Name: columns[0]?.trim() || "",
        Age: parseInt(columns[1]?.trim() || "0", 10),
        Email: columns[2]?.trim() || "",
        Score: parseFloat(columns[3]?.trim() || "0")
      };
    });

    // Output the final JSON array to stdout
    process.stdout.write(JSON.stringify(jsonData, null, 2));
  } catch (error) {
    // In case of unexpected parsing errors, output an empty array to ensure valid JSON
    process.stdout.write('[]');
  }
}

main();